shared/chat/conversation/info-panel/menu/container.js

```diff
@@ -15,15 +22,32 @@ type OwnProps = {
   visible: boolean,
 }
 
+const moreThanOneSubscribedChannel = createCachedSelector(
+  (state, _) => state.chat2.metaMap,
+  (_, teamname) => teamname,
+  (metaMap, teamname) => {
+    console.log('aaaa reselect, metaMap, teamname', metaMap, teamname)
+    let found = 0
+    return metaMap.some(c => {
+      found += c.teamname === teamname ? 1 : 0
+      // got enough
+      if (found === 2) {
+        return true
+      }
+      return false
+    })
+  }
+)((_, teamname) => teamname)
+
 const mapStateToProps = (state: TypedState, {teamname, isSmallTeam}: OwnProps) => {
   const yourOperations = Constants.getCanPerform(state, teamname)
   // We can get here without loading canPerform
   const _hasCanPerform = Constants.hasCanPerform(state, teamname)
   const badgeSubscribe = !Constants.isTeamWithChosenChannels(state, teamname)
-  const numberOfSubscribedChannels = Constants.getNumberOfSubscribedChannels(state, teamname)
```

Comment:
this was going through all inbox items several times (once for each team) so it's an O(N^2) on your teams

Reply:
also it ran always , even if we were looking at a small team